bardelborgo/front-end/src/app/transaction-list/transaction-list.component.ts

```diff
@@ -1,22 +1,80 @@
-import { Component, ElementRef, ViewChild, inject } from '@angular/core';
-import { TransactionClient } from '../services/api.service';
+import {
+  Component,
+  ElementRef,
+  HostListener,
+  ViewChild,
+  inject,
+} from '@angular/core';
+import { Transaction, TransactionClient } from '../services/api.service';
 import { AppStateService } from '../services/app-state.service';
 import Swal from 'sweetalert2';
 import { SignalRService } from '../services/signalR.service';
+import { Renderer2 } from '@angular/core';
 
 @Component({
   selector: 'app-transaction-list',
   templateUrl: './transaction-list.component.html',
-  styleUrls: ['./transaction-list.component.scss']
+  styleUrls: ['./transaction-list.component.scss'],
 })
 export class TransactionListComponent {
   state = inject(AppStateService);
   signalHub = inject(SignalRService);
   transactionClient = inject(TransactionClient);
-  @ViewChild('scrollContainer') scrollContainer!: ElementRef;
+  @ViewChild('scrollContainer') scrollContainer!: ElementRef; // prende un elemento nel DOM (via ID)
 
-  ngAfterViewChecked() {
-    this.scrollContainer.nativeElement.scrollTop = this.scrollContainer.nativeElement.scrollHeight;
+  transactions: Transaction[] = [];
+  pageSize = 10;
+  currentPage = 1;
+  lastPage = false;
+
+  constructor(private renderer: Renderer2) {}
+
+  ngAfterViewInit() {
+    this.addScrollListener();
+  }
+
+  addScrollListener() {
+    const scrollableDiv = document.getElementById('scrollContainer');
+    this.renderer.listen(scrollableDiv, 'scroll', (event: Event) => {
```

Comment:
NIT: prova a usare scrollContainer. (elemento) ? dovrebbe esserci un elemnto con lo stesso tipo del document.getElementById

```suggestion
    this.renderer.listen(scrollContainer.element (?), 'scroll', (event: Event) => {
```